stackrox/acs-fleet-manager
# added cache to fleetshard-sync reconciler

## Description
Implemented a cache in fleetshard-sync to make sure ManagedCentrals, that have not been changed since the last reconciliation loop, are not reconciled again.

## Checklist (Definition of Done)


## Test manual

Manual testing:
- Run db, fleet-manager and stackrox operator
- Run fleetshard-sync in debug mode using your IDE
- Set Breakpoint at `fleetshard/pkg/runtime/runtime.go:86` (1) and `fleetshard/pkg/runtime/runtime.go:89` (2)
- Create central resource with POST requests to fleetmanager
- Wait for first reconcilation, it should run into breakpoint 2 
- Next reconcilations should always run into breakpoint 1
- change host field in db table `update dinosaur_requests set host='cachtest1.cluster.local';`
- Next reconcilation should run into breakpoint 2
- Further reconcilations should again run into breakpoint 1

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```
